esmvaltool/diag_scripts/impact/quick_insights.py

```diff
@@ -0,0 +1,210 @@
+import pandas as pd
+import seaborn as sns
+import xarray as xr
+
+from esmvaltool.diag_scripts.shared import (
+    ProvenanceLogger,
+    get_diagnostic_filename,
+    get_plot_filename,
+    group_metadata,
+    run_diagnostic,
+    select_metadata,
+)
+
+logger = logging.getLogger(Path(__file__).stem)
+
+
+def log_provenance(filename, ancestors, caption, cfg):
+    # We'll add some extra code to make sure the output is saved and tracked
+    provenance = {
+        'caption': caption,
+        'domains': ['reg'],
+        'authors': ['kalverla_peter'],
+        'references': ['isenes3'],
+        'ancestors': ancestors,
+    }
+    with ProvenanceLogger(cfg) as provenance_logger:
+        provenance_logger.log(filename, provenance)
+
+
+def make_standard_calendar(xrda: 'xr.DataArray'):
+    """Make sure time coordinate uses the default calendar.
+
+    Workaround for incompatible calendars 'standard' and 'no-leap'.
+    Assumes yearly data.
+    """
+    try:
+        years = xrda.time.dt.year.values
+        xrda['time'] = [datetime(year, 7, 1) for year in years]
+    except TypeError:
+        # Time dimension is 0-d array
+        pass
+    except AttributeError:
+        # Time dimension does not exist
+        pass
+
+
+def load_data(metadata: list):
+    """Load all files from metadata into an Xarray dataset.
+
+    ``metadata`` is a list of dictionaries with dataset descriptors.
+    """
+
+    data_arrays = []
+    identifiers = []
+    ancestors = []
+
+    for infodict in metadata:
+        alias = infodict['alias']
+        input_file = infodict['filename']
+        short_name = infodict['short_name']
+
+        xrds = xr.open_dataset(input_file)
+        xrda = xrds[short_name]
+
+        # Make sure datasets can be combined
+        make_standard_calendar(xrda)
+        redundant_dims = np.setdiff1d(xrda.coords, xrda.dims)
+        xrda = xrda.drop(redundant_dims)
+
+        data_arrays.append(xrda)
+        identifiers.append(alias)
+        ancestors.append(input_file)
+
+    # Combine along a new dimension
+    data_array = xr.concat(data_arrays, dim='dataset')
+    data_array['dataset'] = identifiers
+
+    return data_array, ancestors
+
+
+def area_weighted_mean(data_array: 'xr.DataArray') -> 'xr.DataArray':
+    """Calculate area mean weighted by the latitude."""
+    weights_lat = np.cos(np.radians(data_array.lat))
+    means = data_array.weighted(weights_lat).mean(dim=['lat', 'lon', 'time'])
+
+    return means
+
+
+def calculate_bias(model_data: 'xr.DataArray',
+                   obs_data: 'xr.DataArray') -> 'xr.DataArray':
+    """Calculate area weighted RMSD with respect to (mean of) observations."""
+    if len(obs_data['dataset']) > 1:
+        obs_data = obs_data.mean(dim='dataset')
+    else:
+        obs_data = obs_data.squeeze()
+
+    diff = model_data - obs_data
+    bias = area_weighted_mean(diff**2)**0.5
+
+    bias.attrs = model_data.attrs
+    return bias
+
+
+def plot_scatter(tidy_df, ancestors, cfg):
```

Comment:
The variable name `tidy_df` is not very descriptive in this context. Might as well go for `data` or `dataframe`.

```suggestion
def plot_scatter(dataframe, ancestors, cfg):
```

Reply:
Done